test/draw_test.py

```diff
@@ -1,7 +1,309 @@
+#################################### IMPORTS ###################################
+
 import unittest
-from pygame.tests import test_utils
+
 import pygame
 from pygame import draw
+from pygame.locals import SRCALPHA
+from pygame.tests import test_utils
+
+
+def get_border_values(surface, width, height):
+    """
+    Returns a list containing lists with the values of the surface's
+    borders.
+    """
+    border_top = [surface.get_at((x, 0)) for x in range(width)]
+    border_left = [surface.get_at((0, y)) for y in range(height)]
+    border_right = [
+        surface.get_at((width - 1, y)) for y in range(height)]
+    border_bottom = [
+        surface.get_at((x, height - 1)) for x in range(width)]
+
+    return [border_top, border_left, border_right, border_bottom]
+
+
+class DrawEllipseTest(unittest.TestCase):
+    """
+    Class for testing ellipse().
+    """
+    def test_ellipse(self):
+        """ |tags: ignore|
+        Draws ellipses of differing sizes on surfaces of differing sizes and
+        checks to see if the number of sides touching the border of the surface
+        is correct.
+        """
+        # __doc__ (as of 2008-08-02) for pygame.draw.ellipse:
+
+          # pygame.draw.ellipse(Surface, color, Rect, width=0): return Rect
+          # draw a round shape inside a rectangle
+          # 
+          # Draws an elliptical shape on the Surface. The given rectangle is the
+          # area that the circle will fill. The width argument is the thickness
+          # to draw the outer edge. If width is zero then the ellipse will be
+          # filled.
+          # 
+
+        left_top = [(0, 0), (1, 0), (0, 1), (1, 1)]
+        sizes = [(5, 4)]
+        color = (1, 13, 24, 255)
+
+        def same_size(width, height, border_width):
+            """
+            Test for ellipses with the same size as the surface.
+            """
+            surface = pygame.Surface((width, height))
+
+            draw.ellipse(
+                surface, color, (0, 0, width, height), border_width)            
+
+            # For each of the four borders check if it contains the color
+            borders = get_border_values(surface, width, height)
+            for border in borders:
+                self.assertTrue(color in border)
+
+        def not_same_size(width, height, border_width, left, top):
+            """
+            Test for ellipses that aren't the same size as the surface.
+            """
+            surface = pygame.Surface((width, height))
+
+            draw.ellipse(surface, color, (left, top, width - 1, height - 1),
+                         border_width)
+
+            borders = get_border_values(surface, width, height)
+
+            # Check if two sides of the ellipse are touching the border
+            sides_touching = [
+                color in border for border in borders].count(True)
+            self.assertEqual(sides_touching, 2)
+
+        for width, height in sizes:
```

Comment:
a loop over `sizes` which contains only one element ? bit strange...

Reply:
Thanks for pointing that out! I forgot I had removed the other sizes... added them back in.